### ISSUE TITLE
Pump settings

### ISSUE THREAD

**Issue by gempickfordwaugh** [timomer/HAPP#43]
Under the pump settings add in max TBR % - it's currently suggesting 300% and 400% when Roche accu-chek max (if you don't play with the pump) is 250%.

Presumably another option would be to suggest the absolute value as an extended bolus if it's over the max TBR?

**Comment by timomer** [comment]
Hi

Settings > APS Settings > Max Basal - this is U/hr

Ideally this should be in % is pump works by %

Please let me know if that is not your request?

**Comment by gempickfordwaugh** [comment]
Yes and no.  Presumably you need to know the maximum u/hr rate you can set
as basal (50 u/h for Roche Accu-chek), and also need to set the maximum
TBR% you can set (250% for Roche Accu-chek).  As they are two different
things really?  (the max TBR doesn’t equal the max u/h if your average u/h
is 1 or 2 units, so HAPP can think you can do more in one go than you can)



Presumably the TBRs suggested are on 10% increments and that suits all
pumps (it suits mine/Roche Accu-chek), or would it help to have the
increments as a setting (I think some pumps are on 5% increments?  If so
might be worth including that too?).

**Comment by timomer** [comment]
You are right, it does change per pump - I have been finding this out more and more as requests come in - you really cannot expect anything to match across all pumps unfortunately. 

Max u/hr and basal % - this is the users requested max, so I guess its fine to have one setting. The hardcoded limits will kick in way before the pumps 50 u/h limit. But I want to think about this some more. 

The Roche 250% limit is very limiting with the OpenAPS algorithm, if you can link the pump to the desktop software you can remove that limit

**Comment by gempickfordwaugh** [comment]
Presumably if HAPP is requesting a TBR higher than the % the pump allows,
then as it also gives a specified unit value then this can be delivered by
an extended bolus instead of TBR? (as long as you remember to delete the
current basal value from the total).  If partway through the extended bolus
it tells you to cancel the "TBR" then cancelling an extended bolus will be
the same as cancelling a TBR - you'd have received the same amount over the
same time period so IOB would remain the same?

Just trying to think of ways round it so that its adaptable to many pumps
but without having to increase all options in the settings.

**Comment by timomer** [comment]
yep you are correct, and on some pumps we suggest Extended Bolus for this reason. My only concern here is that things start to get complicated when you try and support different configurations for the same pump, the pump has the correct ability, it just needs to be set. I am starting to think it would be a good idea to have a pump profile that the user can customize for their own use - this will allow the flexibility needed and remove the need for me to maintain it.

**Comment by jroth-haj** [comment]
Hi Tim,
I've a newer medtronic Veo and only can use it in open loop mode. This is OK, but I can't accept the suggestions of HAPP because I can't set thi at the pump.
The TBR suggestions of HAPP are grater then I can set. It's hardcoded max 200% (means +100%)
I suggest to implement this as configurable options for the pump profile.
I also suggest to double check or limit this setting to max xxx% because of security.

The next mission Option may be to set the time steps (e.g. 15 min / 30 min).

Greetings, Jürgen 

Comment changed:

This looks simmilar to #15
